.pre-commit-config.yaml

```
@@ -0,0 +1,12 @@
+exclude: ^caim_base/static/vendor/
+repos:
+  - repo: https://github.com/psf/black
+    rev: 23.7.0
+    hooks:
+      - id: black
+  - repo: https://github.com/rtts/djhtml
+    rev: "3.0.6"
+    hooks:
+      - id: djhtml
+      - id: djcss
+      - id: djjs
```

could we also add flake8, isort, and some other QoL checks:
```suggestion
      - id: djjs
  -   repo: https://github.com/timothycrosley/isort
    rev: 5.12.0
    hooks:
    -   id: isort
        language_version: python3
        args: [
                --multi-line, "3",
                "--trailing-comma",
                "--use-parentheses",
            ]
-   repo: https://github.com/PyCQA/flake8
    rev: 6.0.0
    hooks:
    -   id: flake8
        language_version: python3
        additional_dependencies:
            - flake8-bugbear
            - flake8-comprehensions
            - flake8-debugger
            - flake8-string-format
        exclude: django_app/frontend/views.py
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
    -   id: check-executables-have-shebangs
    -   id: check-yaml
    -   id: end-of-file-fixer
    -   id: trailing-whitespace
    -   id: detect-private-key
    -   id: name-tests-test
        args: [--django]
```

---

Those are good suggestions. After thinking about it a bit further, and to help consolidate the config, I've switched to [Ruff](https://github.com/astral-sh/ruff), which provides most of these features out of the box (and is quite a bit faster).